libr/bin/p/bin_dex.c

```diff
@@ -543,6 +546,28 @@ static void dex_parse_debug_item(RBinFile *binfile, RBinDexObj *bin,
 		}
 		opcode = *(p4++) & 0xff;
 	}
+
+	if (!binfile->sdb_addrinfo) {
+		binfile->sdb_addrinfo = sdb_new0 ();
+	}
+
+	char fileline[128];
+	char offset[64];
+	char *offset_ptr;
+
+	RListIter *iter1;
+	struct dex_debug_position_t *pos;
+	r_list_foreach (debug_positions, iter1, pos) {
+		snprintf (fileline, sizeof (fileline) - 1, "%s|%"PFMT64d, getstr (bin, pos->source_file_idx), pos->line);
+		offset_ptr = sdb_itoa (pos->address + paddr, offset, 16);
+		if (!sdb_add (binfile->sdb_addrinfo, offset_ptr, fileline, 0)) {
+			sdb_set (binfile->sdb_addrinfo, offset_ptr, fileline, 0);
+		}
+		if (!sdb_add (binfile->sdb_addrinfo, fileline, offset_ptr, 0)) {
```

Comment:
wtf? sdb-add only adds if the key doesnt exist. there's no sense in setting the value if it already exist